src/control/jobs/control_jobs.c

```diff
@@ -2060,7 +2060,38 @@ void dt_control_write_sidecar_files()
                                                           FALSE));
 }
 
+static gboolean _same_basefilename(const char *filename, const char *prev_filename)
+{
+  if(!filename || !prev_filename) return FALSE;
+  const char *dot = strrchr(filename, '.');
+  if(!dot) return FALSE;
+  const int length = dot - filename;
+  const char *prev_dot = strrchr(prev_filename, '.');
+  if(!prev_dot) return FALSE;
+  const int prev_length = prev_dot - prev_filename;
+  if(length != prev_length)
+    return FALSE;
+  for(int i = length - 1; i > 0; i--)
+    if(filename[i] != prev_filename[i])
+      return FALSE;
+  return TRUE;
+}
+
+static char *_set_extension(const char *prev_output, const char *filename)
+{
+  const char *dot = strrchr(filename, '.');
+  const char *prev_dot = strrchr(prev_output, '.');
+  const int name_lgth = prev_dot - prev_output;
+  const int ext_lgth = strlen(dot);
+  char *output = malloc(name_lgth + ext_lgth + 1);
+  memcpy(output, prev_output, name_lgth);
+  memcpy(&output[name_lgth], &filename[strlen(filename) - ext_lgth], ext_lgth);
```

Comment:
`... ext_lgth + 1);`

Would copy the ending \0 and then you don't need following line?

Reply:
Yes.